src/api/wallet/composeProvider.ts

```diff
@@ -54,7 +70,11 @@ function providerAsMiddleware(provider: Provider): JsonRpcMiddleware {
 
     provider[sendFName](req, (err: JsonRpcError<unknown>, providerRes: JsonRpcResponse<unknown>) => {
       // forward any error
-      if (err) return end(err)
+      if (err) {
+        sanitizeErrorData(err)
+        return end(err)
+      }
+      sanitizeErrorData(providerRes.error)
```

Comment:
Bit confused here.

What do we expect `providerRes.error` to be when `err === true` and when `err === false`? My confusion stems from line 77, that `sanitizeErrorData` will never run because of line 75

But I'm probably understanding this wrong is all ;)

Reply:
Errors can be handled in two ways
We get Object or we get error prop on response
Need to handle them both, because they both can be indirectly formatted

Reply:
so the return doesn't break line 77? or it's one or the other. either way if you see what I mean and it's fine then you've handled it in which case i say CASE CLOSED

Reply:
We call sanitizer whether before return or in else clause (after)